research-software-directory/RSD-as-a-service
# 115 organisations

# Participating organisations

Fixes #115 

Changes proposed in this pull request:

*  Participating organisation section of edit software page makes user able to add organisations:
    * previously added organisations to RSD
    * search in ROR for new organisation to add
    * manually create new organisations that is not in RSD or ROR

How to test:
* `docker-compose down --volumes`: to remove old version of database
* `docker-compose build`: to build new images
* `docker-compose up`: to start instance
* run data-migration: optionally, migrate data
* login to RSD as professor1 (this is important step)
* create new software and navigate to organisations section
* add organisation:
     *  select organisation from ROR
     *  type name of organisation, CREATE button should appear, you will be able to add new organisation
* logout from RSD and login second time but as professor2 (this is important step)
* create new software
* add organisation:
     *  select organisation you previously added, it should be listed as RSD organisation
     *  select organisation from ROR (not already added)
     *  type name of organisation, CREATE button should appear, you will be able to add new organisation
* editing: you will see edit icon for the organisation you created as user (you are maintainer of these organisations)

![image](https://user-images.githubusercontent.com/9204081/158884849-d30b4336-59e1-4ae8-8c7e-a114a0f31e0f.png)

PR Checklist:
